master/pkg/model/task.go

```diff
@@ -146,6 +146,16 @@ type Allocation struct {
 	StatusCode   *int32  `db:"status_code" bun:"status_code"`
 }
 
+// AllocationWorkspaceRecord is the model for persisting the workspace and experiment
+// information associated with an allocation.
+type AllocationWorkspaceRecord struct {
+	bun.BaseModel `bun:"table:allocation_workspace_info"`
+	AllocationID  AllocationID `db:"allocation_id" bun:"allocation_id,notnull"`
+	ExperimentID  int          `db:"experiment_id" bun:"experiment_id"`
+	WorkspaceID   int          `db:"workspace_id" bun:"workspace_id,notnull"`
+	WorkspaceName string       `db:"workspace_name" bun:"workspace_name,notnull"`
```

Comment:
is there a reason why we need to persist workspace name separately? is it to avoid doing another join on the workspaces table?

if at the time of the allocation/persist, the workspace (let's say ID=1) had name "myworkspace", but when fetching historical records the workspace 1 had changed names to "neworkspacename", is expected behavior to show the old name or the latest?

Reply:
The intention here is to show the old name, this was primarily done to handle cases where the workspace had been deleted